### DISCUSSION TITLE
vscode ext host in web container

### DISCUSSION THREAD

**Discussion by kaisalmen** [CodinGame/monaco-vscode-api#216] [Ideas]
@CGNonofr will vscode-ext-host-server run in a web container (https://webcontainers.io/)? Have you tried that already by any chance?

**Comment by CGNonofr** [comment]
I have no idea! but as soon as you can start a server on a port and connect to it from the client, I don't see any reason for it to not be possible

**Reply by kaisalmen** [reply]
We could run the Python LS or the Pyright extension as-is directly in the browser. Should be a low hanging fruit.
Anything requiring a native env will still not work.

**Reply by CGNonofr** [reply]
what do you mean by

> Anything requiring a native env will still not work.

?

It can run in the vscode extension

**Reply by kaisalmen** [reply]
If you run the ext host in the web container you have a node env, but this is not a real Linux OS or even a container. So, everything only requiring node should work, AFAIK

**Reply by CGNonofr** [reply]
OK now I understand what are those webcontainers

AFAIK, they don't support websockets (because of service worker limitations) so it wouldn't be simple

**Reply by kaisalmen** [reply]
> AFAIK, they don't support websockets (because of service worker limitations) so it wouldn't be simple

I haven't looked into it, yet. And if it is likely not simple, I won't do it now. There other things requiring attention.

A web-assembly based Linux container engine is actually a more desirable approach 🙂  Or a one-line command for spinning up a container locally for the backend/server.
